src/node_jsvmapi_types.h

```diff
@@ -34,4 +36,48 @@ struct napi_property_descriptor {
   void* data;
 };
 
+enum napi_valuetype {
+  // ES6 types (corresponds to typeof)
+  napi_undefined,
+  napi_null,
+  napi_boolean,
+  napi_number,
+  napi_string,
+  napi_symbol,
+  napi_object,
+  napi_function,
+};
+
+enum napi_typedarray_type {
```

Comment:
Just to follow up here, yes, napi_valuetype is meant to mirror JavaScript typeof (which is why function is included as its own variant).

Whether it is actually useful to mirror typeof for NAPIs consumers, I am not sure.